crates/sui-core/src/unit_tests/move_package_upgrade_tests.rs

```diff
@@ -457,6 +457,122 @@ async fn test_upgrade_package_compatible_in_dep_only_mode() {
     );
 }
 
+#[tokio::test]
+async fn test_upgrade_package_add_new_module_in_dep_only_mode_pre_v68() {
+    // Allow new modules in deps-only mode for this test.
+    let _guard = ProtocolConfig::apply_overrides_for_testing(|_, mut config| {
+        config.set_disallow_new_modules_in_deps_only_packages_for_testing(false);
+        config
+    });
+
+    let mut runner = UpgradeStateRunner::new("move_upgrade/base").await;
+
+    let (digest, modules) = build_upgrade_test_modules("dep_only_upgrade_new_module");
+    let effects = runner
+        .upgrade(
+            UpgradePolicy::DEP_ONLY,
+            digest,
+            modules,
+            vec![SUI_FRAMEWORK_PACKAGE_ID, MOVE_STDLIB_PACKAGE_ID],
+        )
+        .await;
+
+    assert!(effects.status().is_ok(), "{:#?}", effects.status());
+}
+
+#[tokio::test]
+async fn test_upgrade_package_add_new_module_in_dep_only_mode() {
+    let mut runner = UpgradeStateRunner::new("move_upgrade/base").await;
+
+    let (digest, modules) = build_upgrade_test_modules("dep_only_upgrade_new_module");
```

Comment:
would it be possible to slice out the new module here, retry the publish, and see that it succeeds?

Reply:
We have that exact test pretty much here: https://github.com/MystenLabs/sui/blob/main/crates/sui-core/src/unit_tests/move_package_upgrade_tests.rs#L573

This is the same package but without `new_module.move`.

Reply:
That is a slightly different from what I am asking. As it stands, it is not guaranteed that the error in this test from the fact that there is a new module. 
If you take the modules list and splice it out, you would guarantee that within this test.

Reply:
Updated so they all start from the valid `dep_only_upgrade`, and then explicitly add or remove files from that.